Source/WebCore/rendering/updating/RenderTreeUpdater.h

```diff
@@ -49,6 +49,7 @@ class RenderTreeUpdater {
     void commit(std::unique_ptr<Style::Update>);
 
     static void tearDownRenderers(Element&);
+    static void tearDownRenderersForShadowRootInsertion(Element&);
     static void tearDownRenderersAfterSlotChange(Element& host);
```

Comment:
maybe these could just use the same function though naming is difficult.

Reply:
We could have exposed an enum for the two different options but I figured an extra method would do. We can always change this in the future if someone has a stronger opinion.